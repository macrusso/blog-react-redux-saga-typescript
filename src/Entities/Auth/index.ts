import * as authConstants from './authConstants';
import * as authActions from './authActions';
import * as authSagas from './authSagas';
import authReducer, { ICommentsState } from './authReducer';

export { authSagas, ICommentsState, authActions, authReducer, authConstants };
