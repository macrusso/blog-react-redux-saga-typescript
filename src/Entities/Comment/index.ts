import * as commentConstants from './commentConstants';
import * as commentSelectors from './commentSelectors';
import * as commentActions from './commentActions';
import * as commentSagas from './commentSagas';
import commentReducer, { ICommentsState } from './commentReducer';
import ICommentPartial from './ICommentPartial';
import IComment from './IComment';

export {
  IComment,
  commentSagas,
  ICommentsState,
  commentActions,
  commentReducer,
  ICommentPartial,
  commentSelectors,
  commentConstants,
};
