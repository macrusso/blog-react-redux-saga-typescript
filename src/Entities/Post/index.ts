import * as postConstants from "./postConstants";
import * as postSelectors from "./postSelectors";
import * as postActions from "./postActions";
import * as postSagas from "./postSagas";
import postReducer from "./postReducer";
import Post from "../../Frontend/Post/PostListItem";
import IPostPartial from "./IPostPartial";
import IPost from "./IPost";
import IPostsState from "./IPostsState";

export { Post, IPost, postSagas, postActions, IPostsState, postReducer, IPostPartial, postSelectors, postConstants };
