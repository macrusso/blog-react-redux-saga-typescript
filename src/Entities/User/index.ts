import * as userConstants from './userConstants';
import * as userSelectors from './userSelectors';
import * as userActions from './userActions';
import * as userSagas from './userSagas';
import userReducer, { IUsersState } from './userReducer';
import IUserAuth from './IUserAuth';
import IUserPartial from './IUserPartial';
import IUser from './IUser';

export {
  IUser,
  IUserAuth,
  userSagas,
  userActions,
  IUsersState,
  userReducer,
  IUserPartial,
  userSelectors,
  userConstants,
};
