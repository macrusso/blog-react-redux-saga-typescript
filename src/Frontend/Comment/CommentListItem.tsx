import React from "react";
import { IComment, IUser } from "../../Entities/";
import { Edit, Delete } from "@material-ui/icons";
import moment from "moment";
import { Typography, withStyles, Chip, Divider, Avatar, CardContent, CardActions, IconButton } from "@material-ui/core";

const styles = {
  root: {
    minWidth: "100%",
    padding: 20,
    margin: "5px 0",
  },
  actions: {
    marginBottom: 0,
    justifyContent: "space-between",
  },
};

interface ICommentListItemProps {
  classes: any;
  loading: boolean;
  currentUser?: IUser;
  comments: IComment[];
  usersLoading: boolean;
  selectedPostId?: string;
  users: { [key: string]: IUser };
  handleOpenEditDialog: (comment: IComment) => void;
  handleOpenDeleteDialog: (comment: IComment) => void;
}

const CommentListItem: React.SFC<ICommentListItemProps> = props => {
  const {
    users,
    classes,
    loading,
    comments,
    currentUser,
    usersLoading,
    selectedPostId,
    handleOpenEditDialog,
    handleOpenDeleteDialog,
  } = props;

  return (
    <>
      {!loading &&
        !usersLoading &&
        selectedPostId &&
        comments.length > 0 &&
        comments.map(comment => (
          <div className={classes.root} key={comment._id}>
            <CardActions className={classes.actions}>
              {comment.userId && (
                <Chip
                  avatar={<Avatar>{users[comment.userId].name[0].toUpperCase()}</Avatar>}
                  label={users[comment.userId].name}
                  className={classes.chip}
                />
              )}
              {currentUser && currentUser._id === comment.userId && (
                <div>
                  <IconButton aria-label="Delete" onClick={() => handleOpenDeleteDialog(comment)}>
                    <Delete />
                  </IconButton>
                  <IconButton aria-label="Edit" onClick={() => handleOpenEditDialog(comment)}>
                    <Edit />
                  </IconButton>
                </div>
              )}
            </CardActions>
            <CardContent>
              <Typography variant="caption">{moment(comment.createdAt).format("dddd, Do MMMM YYYY")}</Typography>

              <Typography variant="body2">{comment.body}</Typography>
            </CardContent>
            <Divider />
          </div>
        ))}
    </>
  );
};

export default withStyles(styles)(CommentListItem);
