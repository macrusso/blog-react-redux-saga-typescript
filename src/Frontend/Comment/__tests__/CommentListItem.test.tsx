import React from "react";
import { BrowserRouter } from "react-router-dom";
import { shallow } from "enzyme";
import CommentListItem from "../CommentListItem";
import { IUser, IComment } from "../../../Entities";

describe("CommentListItem", () => {
  test("Snapshot matches", () => {
    const users: { [key: string]: IUser } = {
      "5bcc554259e0dbfda6ed1a55": {
        name: "string",
        email: "string",
        _id: "5bcc554259e0dbfda6ed1a55",
      },
    };
    const comments: IComment[] = [
      {
        body: "string",
        _id: "5bcc554259e0dbfda6ed1a55",
        postId: "5bcc554259e0dbfda6ed1a55",
        userId: "5bcc554259e0dbfda6ed1a55",
      },
    ];
    const wrapper = shallow(
      <BrowserRouter>
        <CommentListItem
          users={users}
          loading={false}
          comments={comments}
          usersLoading={false}
          currentUser={users[1]}
          handleOpenEditDialog={jest.fn()}
          handleOpenDeleteDialog={jest.fn()}
        />
      </BrowserRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
