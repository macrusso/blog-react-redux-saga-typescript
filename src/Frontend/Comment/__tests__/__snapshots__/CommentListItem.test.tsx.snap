// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentListItem Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserRouter>
    <WithStyles(CommentListItem)
      comments={
        Array [
          Object {
            "_id": "5bcc554259e0dbfda6ed1a55",
            "body": "string",
            "postId": "5bcc554259e0dbfda6ed1a55",
            "userId": "5bcc554259e0dbfda6ed1a55",
          },
        ]
      }
      currentUser={undefined}
      handleOpenDeleteDialog={[MockFunction]}
      handleOpenEditDialog={[MockFunction]}
      loading={false}
      users={
        Array [
          Object {
            "_id": "5bcc554259e0dbfda6ed1a55",
            "email": "string",
            "name": "string",
          },
        ]
      }
      usersLoading={false}
    />
  </BrowserRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(CommentListItem)
        comments={
          Array [
            Object {
              "_id": "5bcc554259e0dbfda6ed1a55",
              "body": "string",
              "postId": "5bcc554259e0dbfda6ed1a55",
              "userId": "5bcc554259e0dbfda6ed1a55",
            },
          ]
        }
        currentUser={undefined}
        handleOpenDeleteDialog={[MockFunction]}
        handleOpenEditDialog={[MockFunction]}
        loading={false}
        users={
          Array [
            Object {
              "_id": "5bcc554259e0dbfda6ed1a55",
              "email": "string",
              "name": "string",
            },
          ]
        }
        usersLoading={false}
      />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "comments": Array [
          Object {
            "_id": "5bcc554259e0dbfda6ed1a55",
            "body": "string",
            "postId": "5bcc554259e0dbfda6ed1a55",
            "userId": "5bcc554259e0dbfda6ed1a55",
          },
        ],
        "currentUser": undefined,
        "handleOpenDeleteDialog": [MockFunction],
        "handleOpenEditDialog": [MockFunction],
        "loading": false,
        "users": Array [
          Object {
            "_id": "5bcc554259e0dbfda6ed1a55",
            "email": "string",
            "name": "string",
          },
        ],
        "usersLoading": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(CommentListItem)
          comments={
            Array [
              Object {
                "_id": "5bcc554259e0dbfda6ed1a55",
                "body": "string",
                "postId": "5bcc554259e0dbfda6ed1a55",
                "userId": "5bcc554259e0dbfda6ed1a55",
              },
            ]
          }
          currentUser={undefined}
          handleOpenDeleteDialog={[MockFunction]}
          handleOpenEditDialog={[MockFunction]}
          loading={false}
          users={
            Array [
              Object {
                "_id": "5bcc554259e0dbfda6ed1a55",
                "email": "string",
                "name": "string",
              },
            ]
          }
          usersLoading={false}
        />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "comments": Array [
            Object {
              "_id": "5bcc554259e0dbfda6ed1a55",
              "body": "string",
              "postId": "5bcc554259e0dbfda6ed1a55",
              "userId": "5bcc554259e0dbfda6ed1a55",
            },
          ],
          "currentUser": undefined,
          "handleOpenDeleteDialog": [MockFunction],
          "handleOpenEditDialog": [MockFunction],
          "loading": false,
          "users": Array [
            Object {
              "_id": "5bcc554259e0dbfda6ed1a55",
              "email": "string",
              "name": "string",
            },
          ],
          "usersLoading": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
